compiler/hash-pipeline/src/lib.rs

```diff
@@ -161,11 +161,11 @@ impl<I: CompilerInterface> Compiler<I> {
 
             // Temporarily swap the settings with a patched settings in order
             // for the prelude bootstrap to run
-            let mut old_settings = std::mem::take(ctx.settings_mut());
+            // let mut old_settings = std::mem::take(ctx.settings_mut());
```

Comment:
Originally, these were here to ensure that the prelude runs in it's own instance of the settings. However, this probably doesn't make anymore sense because it doesn't need run with its own configuration.

But... The prelude pipeline should not run to "build/link" because  the main pipeline instance will deal with this. So, this has to be limited to not being `CompilerStageKind::Codegen`.